lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -382,9 +389,11 @@ def trigger_refresh(self, _):
     def _relation_unit(relation: Relation) -> Optional[Unit]:
         """Return the principal unit for a relation."""
         if relation and relation.units:
+            # TODO: Is the implied order guaranteed?
```

Comment:
Is this really a `# TODO` to leave in the code? Couldn't we just test scaling up and checking the length of `relation.units`?

Reply:
The set of comments here are from my review on #142 that was submitted seconds after it was merged :)

Do you mind if we address this (and add unit tests) in a follow up PR, to avoid further scope creep?

Reply:
I noticed ;) 

If we can split the difference -- scale up and log how many units are in the relation. If there's more than one, let's add handling in a follow-up patch. If there's only one unit in a relation with a subordinate, then the implied order shouldn't matter and we can remove the todo.

Hopefully a fast test and not actually asking for a fix, just a "do we really need TODO this, or can we spend a few minutes checking and move the TODO from a question into an action item to resolve in a follow-up?"

Reply:
Fixed in next commit.